sql/src/main/java/io/crate/planner/Planner.java

```diff
@@ -88,20 +99,33 @@
 import java.util.Locale;
 import java.util.Map;
 import java.util.Set;
+import java.util.function.BooleanSupplier;
 
 @Singleton
 public class Planner extends AnalyzedStatementVisitor<PlannerContext, Plan> {
 
     private static final Logger LOGGER = Loggers.getLogger(Planner.class);
+    static String EXPIRED_LICENSE_ERROR_MESSAGE = "Statement is not allowed - License is expired";
 
     private final ClusterService clusterService;
     private final LogicalPlanner logicalPlanner;
     private final Functions functions;
+    private LicenseService licenseService;
+    private BooleanSupplier hasValidLicense;
 
     private String[] awarenessAttributes;
 
+    private final AllowQueryRelationVisitor allowQueryRelationVisitor;
+
     @Inject
-    public Planner(Settings settings, ClusterService clusterService, Functions functions, TableStats tableStats) {
+    public Planner(Settings settings, ClusterService clusterService, Functions functions, TableStats tableStats, LicenseService licenseService) {
+        this(settings, clusterService, functions, tableStats, () -> false);
+        this.licenseService = licenseService;
+        this.hasValidLicense = licenseService::hasValidLicense;
```

Comment:
The license might arrive at certain nodes at a later time - because it arrives via a cluster state change (if it is generated at startup / configured using the `SET` statement) or recovered from the local state. 
Either way, I think we need to implement the observer pattern and have the `LicenseService` notify the interested parties (in our case the `Planner`) that the license is configured/changed - and when the `Planner` is notified of a license change it can adjust the value of the `hasValidLicense` flag accordingly.

Reply:
I just realised it's a method reference. Please ignore this @smakalias